src/api/java/com/minecolonies/api/entity/ai/statemachine/basestatemachine/BasicStateMachine.java

```diff
@@ -92,15 +92,11 @@ public void removeTransition(final T transition)
     {
         if (transition instanceof IStateMachineEvent)
         {
-            final List<T> temp = new ArrayList<>(eventTransitionMap.get(((IStateMachineEvent<?>) transition).getEventType()));
-            temp.remove(transition);
-            eventTransitionMap.put(((IStateMachineEvent<?>) transition).getEventType(), temp);
+            eventTransitionMap.get(((IStateMachineEvent<?>) transition).getEventType()).removeIf(t -> t == transition);
```

Comment:
Are we sure this can not produce an NPE?
Would a getOrDefault(..., Collections.emptyList()) not be better here just to be sure.

Reply:
ye it would indeed cause and NPE, thought that too when writing it and just left it to crash as sth must be wrong when trying to remove one that isnt added at all